### PR TITLE
core: frontend: Add SDP file download for UDP streams

### LINKED ISSUE #1195: camera-manager: provide SDP files for video streams
### Current behaviour

Companion Software used to provide an SDP file for easily connecting to a video stream from applications like VLC. BlueOS does not do this, which is inconvenient for people who are used to using it.

### Expected or desired behaviour

Provide links / easy access to sdp files, ideally for all video streams, but at least for those with H264 encoding.

### Prerequisites

- [X] I have checked to make sure that a similar request has not already been filed or fixed.

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**
Closes #1195

image

### CODE REVIEW

**Review by ES-Alexander — APPROVED** [review]
LGTM :-)

Presumably the actual SDP is generated by mavlink-camera-manager and this part is just the frontend fetching it? :-)

**Comment by joaoantoniocardoso** [comment]
> LGTM :-)
> 
> Presumably the actual SDP is generated by mavlink-camera-manager and this part is just the frontend fetching it? :-)

yep

**Comment by ES-Alexander** [comment]
Nice! :-)

File: core/frontend/src/store/video.ts [hunk 1]
```diff
@@ -169,6 +171,36 @@ class VideoStore extends VuexModule {
...
+        if (response.status === 200) {
+          const blob = new Blob([response.data], { type: 'application/sdp' })
+          const url = URL.createObjectURL(blob)
+          const a = document.createElement('a')
+          a.href = url
+          a.download = `${stream.video_and_stream.name}.sdp`
+          document.body.appendChild(a)
+          a.click()
+          document.body.removeChild(a)
+          URL.revokeObjectURL(url)
```
Comment (patrickelectric) [inline core/frontend/src/store/video.ts:196]: You should use the [file-saver](https://www.npmjs.com/package/file-saver) library that's already available on the project

**Review by patrickelectric — CHANGES_REQUESTED** [review]

**Review by patrickelectric — CHANGES_REQUESTED** [review]
Not working here
image

image

**Comment by patrickelectric** [comment]
I can download the SDP file, but it's not running via VLC, with the network configuration pointing to the SDP URL and also once downloaded the SDP file and opening it in VLC.

**Comment by joaoantoniocardoso** [comment]
> I can download the SDP file, but it's not running via VLC, with the network configuration pointing to the SDP URL and also once downloaded the SDP file and opening it in VLC.

my VLC is broken, but it is working with mpv, can you test it?

**Comment by joaoantoniocardoso** [comment]
> > I can download the SDP file, but it's not running via VLC, with the network configuration pointing to the SDP URL and also once downloaded the SDP file and opening it in VLC.
> 
> my VLC is broken, but it is working with mpv, can you test it?

Fixed my VLC, its playing the downloaded SDP file.

Can you post here the content of the file you are getting from mavlink-camera-manager?

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**